react/AutoCompletedFields.js

```diff
@@ -1,21 +1,35 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import pickBy from 'lodash/pickBy'
-import flow from 'lodash/flow'
+import reduce from 'lodash/reduce'
 import { compose } from 'recompose'
 
 import AddressShapeWithValidation from './propTypes/AddressShapeWithValidation'
 import AddressSummary from './AddressSummary'
-import { removeValidation, removeField } from './transforms/address'
+import { removeValidation } from './transforms/address'
 import { injectRules } from './addressRulesContext'
 import { injectAddressContext } from './addressContainerContext'
 
 const IRRELEVANT_FIELDS = ['country', 'geoCoordinates']
 
-const removeAutoCompletedFields = flow([
-  (address) => removeField(address, 'postalCodeAutoCompleted'),
-  (address) => removeField(address, 'geolocationAutoCompleted'),
-])
+const removeAutoCompletedFields = (address) => {
+  return reduce(
+    address,
+    (newAddress, prop, propName) => {
+      if (
+        address[propName].postalCodeAutoCompleted ||
+        address[propName].geolocationAutoCompleted
+      ) {
+        return newAddress
+      }
+
+      newAddress[propName] = { ...prop }
+
+      return newAddress
+    },
+    {}
+  )
+}
```

Comment:
How about using `filter` instead? I think it would be more readable

Reply:
Something like this?

```
  return Object.entries(address)
    .filter(
      ([_, value]) =>
        !value.geolocationAutoCompleted && !value.postalCodeAutoCompleted
    )
    .reduce((newAddress, [fieldName, fieldValue]) => {
      newAddress[fieldName] = fieldValue
      return newAddress
    }, {})
```

Reply:
What do you guys think? @wdsrocha @lucasecdb

Reply:
I'd prefer to use `Object.fromEntries` at the end in place of the reducer

Reply:
Same as Lucas. My bad, I could've provided an implementation example. With `fromEntries`, the code would look like this, right? (without formatting :v)
```suggestion
const removeAutoCompletedFields = Object.fromEntries(Object.entries(address)
    .filter(
      ([_, value]) =>
        !value.geolocationAutoCompleted && !value.postalCodeAutoCompleted
    )})
```

Reply:
I'm just seeing this comment now (fack github), but, yeah, it will look like this.